src/fat_beagle.hpp

```diff
@@ -137,14 +138,19 @@ std::vector<TOut> FatBeagleParallelize(
   }
   Assert(tree_collection.TreeCount() == param_matrix.rows(),
          "We param_matrix needs as many rows as we have trees.");
-  TaskProcessor<FatBeagle *, size_t> task_processor(
-      std::move(fat_beagle_queue), std::move(tree_number_queue),
-      [&results, &tree_collection, &param_matrix, &rescaling, &f](FatBeagle *fat_beagle,
-                                                                  size_t tree_number) {
-        fat_beagle->SetParameters(param_matrix.row(tree_number));
-        fat_beagle->SetRescaling(rescaling);
-        results[tree_number] = f(fat_beagle, tree_collection.GetTree(tree_number));
-      });
+
+  std::unique_ptr<TaskProcessor<FatBeagle *, size_t>> task_processor =
+    std::make_unique<TaskProcessor<FatBeagle *, size_t>>(
+    std::move(fat_beagle_queue), std::move(tree_number_queue),
+    [&results, &tree_collection, &param_matrix, &rescaling, &f](FatBeagle *fat_beagle,
+                                                                size_t tree_number) {
+      fat_beagle->SetParameters(param_matrix.row(tree_number));
+      fat_beagle->SetRescaling(rescaling);
+      results[tree_number] = f(fat_beagle, tree_collection.GetTree(tree_number));
+    });
+    
+  task_processor->Wait();
```

Comment:
Note that we need to update the docs at the top of this file now that we need to explicitly Wait.

Reply:
Should have thought of it sooner.  There's no reason we can't move Wait() to the end of the Constructor rather than the Destructor is there?  It still passes the test case with this reorganization.

Reply:
Interesting idea! And this doesn't have the problem you found before because at the end of the constructor we aren't going out of scope?

Reply:
Correct.

Reply:
Love it